common/src/main/resources/common/common.bom

```diff
@@ -172,3 +172,52 @@ brooklyn.catalog:
           launch.command: |
             haproxy -f haproxy.conf -f servers.conf -c &&
               haproxy -D -f haproxy.conf -f servers.conf -sf $(cat ${PID_FILE})
+
+        brooklyn.initializers:
+        - type: org.apache.brooklyn.core.effector.ssh.SshCommandEffector
+          brooklyn.config:
+            name: configureHaproxy
+            description: |
+              Generate HAProxy configuration from endpoint list
+            shell.env:
+              HAPROXY_PROTOCOL: $brooklyn:config("haproxy.protocol")
+              HAPROXY_ENDPOINTS: $brooklyn:attributeWhenReady("haproxy.endpoints")
+            command: |
+              (
+              flock -o 9
+              cat > ${RUN_DIR}/servers.conf <<-EOF
+              backend servers
+              EOF
+              if [ "${HAPROXY_PROTOCOL}" == "http" ]; then
+              cat >> ${RUN_DIR}/servers.conf <<-EOF
+                mode http
+                option httpchk
+              EOF
+              fi
+              cat >> ${RUN_DIR}/servers.conf <<-EOF
+                balance roundrobin
+              EOF
+              for endpoint in $(echo ${HAPROXY_ENDPOINTS} | tr ',' ' ') ; do
+                if [ "${HAPROXY_PROTOCOL}" == "http" ]; then
+                  echo "  server master$((++n)) ${endpoint} check" >> ${RUN_DIR}/servers.conf
+                else
+                  echo "  server manager$((++n)) ${endpoint} maxconn 32 ssl ca-file ${RUN_DIR}/ca.pem crt ${RUN_DIR}/cert.pem" >> ${RUN_DIR}/servers.conf
+                fi
+              done
+              OLD_PID=$(cat ${RUN_DIR}/pid.txt)
+              haproxy -D -f ${RUN_DIR}/haproxy.conf -f ${RUN_DIR}/servers.conf -st $OLD_PID
+              # wait for the old process to die before leaving the critical section
+              while test -d "/proc/$OLD_PID" 2> /dev/null; do sleep 1; done;
+              flock -u 9
+              ) 9>>/tmp/configureSwarm.lock
```

Comment:
Should not reference Swarm, and should probably also have entity ID embedded, to deal with running multiple HAProxy instances on the same server (with different ports/protocols, for example) so add an environment variable `ENTITY_ID: $brooklyn:attributeWhenReady("entity.id")` and use `/tmp/configure.${ENTITY_ID}.lock` as the lock file.